base_import_async/models/base_import_async.py

```diff
@@ -189,49 +183,47 @@ def split_file(session, model_name, translated_model_name,
 class BaseImportConnector(TransientModel):
     _inherit = 'base_import.import'
 
-    def do(self, cr, uid, res_id, fields, options, dryrun=False, context=None):
+    def do(self, fields, options, dryrun=False):
         if dryrun or not options.get(OPT_USE_CONNECTOR):
             # normal import
             return super(BaseImportConnector, self).do(
-                cr, uid, res_id, fields, options, dryrun=dryrun,
-                context=context)
+                fields, options, dryrun=dryrun)
 
         # asynchronous import
-        (record,) = self.browse(cr, uid, [res_id], context=context)
         try:
-            data, import_fields = self._convert_import_data(
-                record, fields, options, context=context)
-        except ValueError, e:
+            data, import_fields = self._convert_import_data(fields, options)
+            # Parse date and float field
+            data = self._parse_import_data(data, import_fields, options)
+        except ValueError, error:
```

Comment:
This line would be more idiomatic of recent Python versions as:

```
    except ValueError as error:
```

Or, if you want to reduce the difference between the versions:

```
    except ValueError as e:
```

So that you can get rid of the line difference below (i.e. `'message': unicode(error),`).